Qt35.py

```diff
@@ -0,0 +1,318 @@
+
+
+class Tide(object):
+    def __init__(self, filename, time1=datetime.datetime(2000, 1, 1, 00, 0),
+                 time2=datetime.datetime(2016, 10, 16, 5, 0)):
+        self.data = {}
+        self.year = {}
+        self.month = {}
+        self.day = {}
+        for s in xlrd.open_workbook(filename=filename).sheet_names():
+            try:
+                self.tide_sheet = pandas.read_excel(filename, sheetname=s)
+                if 'tide' in self.tide_sheet.columns.values:
+                    if 'time' in self.tide_sheet.columns.values:
+                        pass
+                    else:
+                        say_out('Please Check Your Format of the Sheet')
+            except:
+                say_out('Please Check Your sheets name of the File')
+
+            temp_data = self.tide_sheet
+
+            t = []
+            if isinstance(temp_data.time[1], pandas.tslib.Timestamp):
+                t = temp_data.time
+            else:
+                for x in temp_data.time:
+                    t.append(dateutil.parser.parse(x))
+
+            temp_data['format_time'] = t
+            temp_data['tide_init'] = temp_data.tide
+            temp_data.tide = process(temp_data.tide)
+            temp_data.index = temp_data['format_time']
+
+            temp_data = temp_data.ix[temp_data.format_time <= time2]
+            temp_data = temp_data.ix[temp_data.format_time >= time1]
+
+            temp_data['if_min'] = np.r_[True, temp_data.tide.values[1:] < temp_data.tide.values[:-1]] & np.r_[
+                temp_data.tide.values[:-1] < temp_data.tide.values[1:], True]
+            temp_data['if_max'] = np.r_[True, temp_data.tide.values[1:] > temp_data.tide.values[:-1]] & np.r_[
+                temp_data.tide.values[:-1] > temp_data.tide.values[1:], True]
+            temp_data.set_value(temp_data.index.max(), 'if_min', False)
+            temp_data.set_value(temp_data.index.max(), 'if_max', False)
+            temp_data.set_value(temp_data.index.min(), 'if_max', False)
+            temp_data.set_value(temp_data.index.min(), 'if_min', False)
+
+            temp_data['diff'] = None
+
+            temp_data['raising_time'] = None
+            temp_data['ebb_time'] = None
+            for x in temp_data.ix[temp_data.if_max == True].format_time:
+                try:
+                    temp_data.loc[x, 'raising_time'] = x - temp_data[
+                        (temp_data.if_min == True) & (temp_data.format_time < x)].format_time.max()
+                    temp_data.loc[x, 'diff'] = temp_data.loc[x, 'tide'] - temp_data.loc[temp_data[
+                                                                                            (
+                                                                                                temp_data.if_min == True) & (
+                                                                                                temp_data.format_time < x)].format_time.max(), 'tide']
+                except:
```

Comment:
- To catch any exception use the `except Exception`, instead of `except BaseException` or just `except`.